deep_agent/src/personalization/repository.py

```
@@ -180,6 +180,25 @@ async def upsert_rule(
+            cur = await conn.execute(
+                "DELETE FROM user_memories WHERE user_id = %s",
+                (user_id,),
+            )
+            await conn.commit()
+            count: int = cur.rowcount
+        if count > 0:
+            try:
+                from deep_agent.src.cache.personalization_cache import invalidate
+
+                await invalidate(user_id)
+            except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)